tests/src/drivers/test_driver_key_management.c

```diff
@@ -56,6 +56,88 @@ const uint8_t mbedtls_test_driver_ecdsa_pubkey[65] =
       0xbc, 0x25, 0x16, 0xc3, 0xd2, 0x70, 0x2d, 0x79,
       0x2f, 0x13, 0x1a, 0x92, 0x20, 0x95, 0xfd, 0x6c };
 
+
+/*
+ * This macro returns the base size for the key context when SE does not support storage.
+ * It is the size of the metadata that gets added to the wrapped key.
+ * In its test functionality the metadata is just some padded prefixing to the key.
+ */
+#define TEST_DRIVER_KEY_CONTEXT_BASE_SIZE  PSA_CRYPTO_TEST_DRIVER_OPAQUE_PAD_PREFIX_SIZE
+
+
+size_t mbedtls_test_opaque_size_function(
+    const psa_key_type_t key_type,
+    const size_t key_bits )
+{
+    size_t key_buffer_size = 0;
+
+    key_buffer_size = PSA_EXPORT_KEY_OUTPUT_SIZE( key_type, key_bits );
+    if( key_buffer_size == 0 )
+        return( key_buffer_size );
+    /* Include spacing for base size overhead over the key size
+     * */
+    key_buffer_size += TEST_DRIVER_KEY_CONTEXT_BASE_SIZE;
+    return( key_buffer_size );
+}
+
+size_t mbedtls_test_opaque_get_base_size()
+{
+    return TEST_DRIVER_KEY_CONTEXT_BASE_SIZE;
+}
+
+/*
+ * The wrap function mbedtls_test_opaque_wrap_key pads and wraps the clear key.
+ * It expects the clear and wrap buffers to be passed in.
+ * key_buffer_size is the size of the clear key to be wrapped.
+ * wrap_buffer_size is the size of the output buffer wrap_key.
+ * The argument key_buffer_length is filled with the wrapped key_size on success.
+ * */
+static psa_status_t mbedtls_test_opaque_wrap_key(
+    const uint8_t *key_buffer,
+    size_t key_buffer_size,
+    uint8_t *wrap_key,
+    size_t wrap_buffer_size,
+    size_t *key_buffer_length )
+{
+   size_t opaque_key_base_size = mbedtls_test_opaque_get_base_size();
+   uint64_t prefix = PSA_CRYPTO_TEST_DRIVER_OPAQUE_PAD_PREFIX;
+   if( key_buffer_size + opaque_key_base_size > wrap_buffer_size )
+      return( PSA_ERROR_BUFFER_TOO_SMALL );
+   /* Write in the opaque pad prefix */
+   memcpy( wrap_key, &prefix, opaque_key_base_size);
+   wrap_key += opaque_key_base_size;
+   *key_buffer_length = key_buffer_size + opaque_key_base_size;
+   while( key_buffer_size-- )
+      wrap_key[key_buffer_size] = key_buffer[key_buffer_size] ^ 0xFF;
+   return( PSA_SUCCESS );
+}
+
+/*
+ * The unwrap function mbedtls_test_opaque_unwrap_key removes a pad prefix and unwraps
+ * the wrapped key. It expects the clear and wrap buffers to be passed in.
+ * wrapped_key_buffer_size is the size of the wrapped key,
+ * key_buffer_size is the size of the output buffer clear_key.
+ * The argument key_buffer_length is filled with the unwrapped(clear) key_size on success.
+ * */
+static psa_status_t mbedtls_test_opaque_unwrap_key(
+    const uint8_t *wrapped_key,
+    size_t wrapped_key_buffer_size,
```

Comment:
https://github.com/ARMmbed/mbedtls/pull/4766#discussion_r694574870, We're running multiple passes over naming here. 
Would be great if we don't run in circles here :).

Reply:
The convention for API functions is `xxx_length` for the actual size of the data and `xxx_size` for a buffer size which can be more than the size. Internal functions don't absolutely have to obey this convention, but it makes the code base easier to read if they do.

Additionally, `key_size` or `adjective_key_size` should be avoided because “key size” can have two meanings: the size of the representation (disambiguated by e.g. `key_material_size`, `key_data_size`, `key_buffer_size`), or size of some value that reflects the strength of the key (which we always express in bits, disambiguated as `key_bits`).

Ronald's first comment was about avoiding the `key_size` ambiguity, but `length` is a better name since the parameter is the size of the actual data and not the size of some buffer which can be larger than the data.